<?xml version="1.0"?>
<launch>
	<rosparam file="$(find maps)/yaml/d_kan_indoor.yaml" />
	<!-- <rosparam file="$(find maps)/yaml/d_kan_around.yaml" /> -->
	<!-- <rosparam file="$(find maps)/yaml/tkb_2018_kakunin.yaml" /> -->
	<!-- <rosparam file="$(find maps)/yaml/tkb_2018_kakunin_L.yaml" /> -->
	<!-- <rosparam file="$(find maps)/yaml/ikuta_all.yaml" /> -->

	<!-- 1:indoor 2:around other:tkb -->
	<param name="map_mode" value="1" />
	
	<node pkg="kari_localization" type="map_alignment_add_height" name="alignment"  >
		<param name="laser_voxel_size" value="0.3" />
		<param name="map_voxel_size" value="0.3" />
		<param name="map_limit" value="20.0" />
		<param name="laser_limit" value="20.0" />
		<!-- <param name="lidar_topic" value="/cloud/lcl" /> -->
		<!-- <param name="lidar_topic" value="/velodyne_obstacles_after" /> -->
		<param name="lidar_topic" value="/velodyne_obstacles" />

		<!--tkb -->
		<!-- <param name="map_limit" value="50.0" /> -->
		<!-- <param name="laser_limit" value="50.0" /> -->
	
	</node>

	<!-- <node pkg="kari_localization" type="get_height" name="get_height" /> -->
	<node pkg="kari_localization" type="pc2transform4map" name="get_height" output="screen"/>

	<!-- <node pkg="kari_localization" type="pcd_loader_" name="pcd_loader" /> -->

	<node name="imu_complement" pkg="kari_complement" type="imu_complement_add_height" >
		<param name="header_frame" value="/map" />	
		<param name="child_frame" value="/matching_base_link" />
		<param name="init_x" value="0.0" />
		<param name="init_y" value="0.0" />
		<param name="init_yaw" value="0.0" />
		<!-- 0:odom 1:imu 2:ndt 3:ekf -->
		<param name="estimate_result" value="3" />
		<param name="dyaw/drift" value="0" />
		
		<remap from="/imu/data" to="/imu/data/calibrated" />
		<!-- <remap from="/odom" to="/tinypower/odom" /> -->
	</node>


    <node pkg="ekf" type="sq_NDT_odo" name="sq_NDT_odo" >
        <rosparam file="$(find params)/run_test/sq_ekf_sigs.yaml" />
		<param name="init_x" value="0.0" />
		<param name="init_y" value="0.0" />
		<param name="init_yaw" value="0.0" />
		<remap from="/imu/data" to="/imu/data/calibrated" />
		<!-- <remap from="/odom" to="/tinypower/odom" /> -->
	</node>

	<node name="waypoint_manager" pkg="waypoint_manager" type="waypoint_manager" output="screen"/>

</launch>
